profile/profile.go

```diff
@@ -80,35 +81,56 @@ func SaveMemProfile(profileDir, identifier string) error {
 	return nil
 }
 
-// StartContinuousProfiling starts profiling in a gothread, logging at
-// intervals various metrics such as the number of gothreads and some memory
-// statistics.
-func StartContinuousProfile(profileDir string) {
+// StartContinuousProfiling will continuously print statistics about the cpu
+// usage, memory usage, and runtime stats of the program.
+func StartContinuousProfile(profileDir string, cpuProfiling bool, memProfiling bool) {
+	// Create the folder for all of the profiling results.
 	err := os.MkdirAll(profileDir, 0700)
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
 
-	// Create a logger to infrequently log performance statistics.
+	// Continuously log statistics about the running Sia application.
 	go func() {
-		// Create a logger for the goroutine.
-		logFile, err := os.OpenFile(filepath.Join(profileDir, "continuousProfiling.log"), os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0660)
+		// Create the logger.
+		log, err := persist.NewLogger(filepath.Join(profileDir, "continuousProfiling.log"))
 		if err != nil {
-			fmt.Println("Goroutine logging failed:", err)
+			fmt.Println("Profile logging failed:", err)
 			return
 		}
-		log := log.New(logFile, "", log.Ldate|log.Ltime|log.Lmicroseconds|log.Lshortfile)
-		log.Println("Continuous profiling started.")
 
-		// Infinite loop to print out the goroutine count.
-		sleepTime := time.Second * 3
+		// Collect statistics in an infinite loop.
+		identifier := "continious-profiling"
+		sleepTime := time.Second * 20
 		for {
 			// Sleep for an exponential amount of time each iteration, this
 			// keeps the size of the log small while still providing lots of
 			// information.
 			time.Sleep(sleepTime)
 			sleepTime = time.Duration(1.5 * float64(sleepTime))
+
+			if cpuProfiling {
+				// If performing cpu profiling, stop the previous profiler so
+				// that the data will be dumped to disk, then start a new cpu
+				// profiler for this next run.
+				StopCPUProfile()
+				err = StartCPUProfile(profileDir, identifier)
+				if err != nil {
+					fmt.Println("CPU profiling failed:", err)
+					return
+				}
+			}
+
+			if memProfiling {
+				// If performing memory profiling, grab a profile of the
+				// current memory structure.
+				err = SaveMemProfile(profileDir, identifier)
+				if err != nil {
+					fmt.Println("memory logging failed:", err)
+				}
+			}
+
```

Comment:
I extended the profiling capabilities a bit, though honestly I haven't found these extensions to be that useful